src/org/exist/util/ExistSAXParserFactory.java

```diff
@@ -53,7 +53,9 @@ public static SAXParserFactory getSAXParserFactory(final String className) {
 
         } catch (final ClassNotFoundException ex) {
             // quick escape
-            LOG.debug(className + ": " + ex.getMessage(), ex);
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(className + ": " + ex.getMessage(), ex);
```

Comment:
probably is with only

`LOG.debug("{}: {}", className, ex.getMessage(),ex)` almost as fast...?

Reply:
Nope. Without the `if` getMessage is always evaluated even when debug logging is disabled.